Tools/TestWebKitAPI/cocoa/TestNavigationDelegate.mm

```diff
@@ -100,7 +100,7 @@ - (void)webView:(WKWebView *)webView didReceiveAuthenticationChallenge:(NSURLAut
 
 - (void)_webView:(WKWebView *)webView navigation:(WKNavigation *)navigation didSameDocumentNavigation:(_WKSameDocumentNavigationType)navigationType
 {
-    if (_didSameDocumentNavigation)
+    if (_didSameDocumentNavigation && navigationType != _WKSameDocumentNavigationTypeAnchorNavigation)
```

Comment:
TestNavigationDelegate's job is to be thin and prevent each test from needing its own class declaration.  I think this logic would be better placed in the test.